### ISSUE TITLE
Makefile behaves differently on different systems

### ISSUE THREAD

**Issue by SoniEx2** [WebAssembly/wabt#1558]
Trying to use the repo makefile just doesn't work. It's very inconsistent and different GNU make versions behave differently, and many tests fail when using it.

On 2 different systems, the makefile was not able to run the c-api examples. On a third system, it ran the tests instead of the examples, but the wat2wasm tests failed because it couldn't find wat2wasm somehow?

... Nevertheless, using cmake directly seemed to make it all work. So...? This is weird tbh.

**Comment by binji** [comment]
Yeah, the `Makefile` is weird and mostly only convenient for me to hack multiple configurations on wabt. It probably should just be removed or made less prominent.
